docs/source/lightning_cli.rst

```diff
@@ -0,0 +1,341 @@
+format and for the example above would look as follows:
+
+.. code-block:: bash
+
+    $ python trainer.py --print_config
+    model:
+      decoder_layers:
+      - 2
+      - 4
+      encoder_layers: 12
+    trainer:
+      accelerator: null
+      accumulate_grad_batches: 1
+      amp_backend: native
+      amp_level: O2
+      ...
+
+Note that there is a section for each class (model and trainer) including all
+the init parameters of the class. This grouping is also used in the formatting
+of the help shown previously.
+
+
+Trainer Callbacks and arguments with class type
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A very important argument of the
+:class:`~pytorch_lightning.trainer.trainer.Trainer` class is the
+:code:`callbacks`. In contrast to other more simple arguments which just require
+numbers or strings, :code:`callbacks` expects a list of instances of subclasses
+of :class:`~pytorch_lightning.callbacks.Callback`. To specify this kind of
+argument in a config file, each callback must be given as a dictionary including
+a :code:`class_path` entry with an import path of the class, and optionally an
+:code:`init_args` entry with arguments required to instantiate it. Therefore, a
+simple configuration file example that defines a couple of callbacks is the
+following:
+
+.. code-block:: yaml
+
+    trainer:
+      callbacks:
+        - class_path: pytorch_lightning.callbacks.EarlyStopping
+          init_args:
+            patience: 5
+        - class_path: pytorch_lightning.callbacks.LearningRateMonitor
+          init_args:
+            ...
+
+Similar to the callbacks, any arguments in
+:class:`~pytorch_lightning.trainer.trainer.Trainer` and user extended
+:class:`~pytorch_lightning.core.lightning.LightningModule` and
+:class:`~pytorch_lightning.core.datamodule.LightningDataModule` classes that
+have as type hint a class can be configured the same way using
+:code:`class_path` and :code:`init_args`.
+
+
+Multiple models and/or datasets
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the previous examples :class:`~pytorch_lightning.utilities.cli.LightningCLI`
+works only for a single model and datamodule class. However, there are many
+cases in which the objective is to easily be able to run many experiments for
+multiple models and datasets. For these cases the tool can be configured such
+that a model and/or a datamodule is specified by an import path and init
+arguments. For example, with a tool implemented as:
+
+.. testcode::
+
+    from pytorch_lightning.utilities.cli import LightningCLI
+
+    cli = LightningCLI(
+        MyModelBaseClass,
+        MyDataModuleBaseClass,
+        subclass_mode_model=True,
+        subclass_mode_data=True
+    )
+
+A possible config file could be as follows:
+
+.. code-block:: yaml
+
+    model:
+      class_path: mycode.mymodels.MyModel
+      init_args:
+        decoder_layers:
+        - 2
+        - 4
+        encoder_layers: 12
```

Comment:
To modify these from the command line, would I have to do `--model.init_args.encoder_layers 13`?

Reply:
Yes

Reply:
Would it make sense to forward `--model.encoder_layers` to `--model.init_args.encoder_layers` ? IMO, it would.

Reply:
I am not sure if there is a simple/clean way to implement this, but I will think about it. Could you please create a feature request for this in jsonargparse.

Reply:
I will mark this as resolved since it is not related to the code in pytorch-lightning. This would need to be done in jsonargparse.